File: String to Integer (atoi).cpp
// Leetcode Problem 8: String to Integer (atoi)
// C++ Code
#include <limits.h>
#include <string>

class Solution {
public:
    int myAtoi(std::string s) {
        int i = 0;
        // Skip leading whitespace
        while (s[i] == ' ') {
            i++;
        }

        // Check for sign
        int sign = 1;
        if (s[i] == '-') {
            sign = -1;
            i++;
        } else if (s[i] == '+') {
            i++;
        }

        // Convert digits to integer
        long long result = 0;
        while (i < s.length() && std::isdigit(s[i])) {
            result = result * 10 + (s[i] - '0');
            // Check for overflow
            if (result > INT_MAX) {
                return sign == 1 ? INT_MAX : INT_MIN;
            }
            i++;
        }

        result *= sign;
        return static_cast<int>(result);
    }
};

File: String to Integer (atoi).py
# Leetcode Problem 8: String to Integer (atoi)
# Python Code
class Solution:
    def myAtoi(self, s: str) -> int:
        if not s:
            return 0

        s = s.strip()
        if not s:
            return 0

        sign = 1
        if s[0] == '-':
            sign = -1
            s = s[1:]
        elif s[0] == '+':
            s = s[1:]

        result = 0
        for c in s:
            if not c.isdigit():
                break
            result = result * 10 + int(c)

        result *= sign
        if result > 2**31 - 1:
            return 2**31 - 1
        if result < -2**31:
            return -2**31

        return result